Zend/zend_compile.c

```diff
@@ -5734,10 +5735,15 @@ void zend_compile_params(zend_ast *ast, zend_ast *return_type_ast) /* {{{ */
 			default_node.op_type = IS_CONST;
 			zend_const_expr_to_zval(&default_node.u.constant, default_ast);
 			CG(compiler_options) = cops;
+			have_optional_param = 1;
 		} else {
 			opcode = ZEND_RECV;
 			default_node.op_type = IS_UNUSED;
 			op_array->required_num_args = i + 1;
+			if (have_optional_param) {
+				zend_error(E_COMPILE_WARNING, "Required parameter $%s follows optional parameter",
```

Comment:
Maybe the message should also say this will become an error in a later PHP version. (If that is the plan)